src/object.d

```diff
@@ -3241,6 +3241,226 @@ template RTInfo(T)
+    static if ((is(U1 == float) && is(U2 == float))
+        || (is(U1 == double)   && is(U2 == double))
+        || (is(U1 == real)     && is(U2 == real))
+        || (is(U1 == ifloat)   && is(U2 == ifloat))
+        || (is(U1 == idouble)  && is(U2 == idouble))
+        || (is(U1 == ireal)    && is(U2 == ireal))
+        || (is(U1 == cfloat)   && is(U2 == cfloat))
+        || (is(U1 == cdouble)  && is(U2 == cdouble))
+        || (is(U1 == creal)    && is(U2 == creal)))
+    {
+        template Floating(T)
+        if (is(T == cfloat) || is(T == cdouble) || is(T == creal))
+        {
+          pure nothrow @safe:
+            int compare(T f1, T f2)
+            {
+                int result;
+
+                if (f1.re < f2.re)
+                    result = -1;
+                else if (f1.re > f2.re)
+                    result = 1;
+                else if (f1.im < f2.im)
+                    result = -1;
+                else if (f1.im > f2.im)
+                    result = 1;
+                else
+                    result = 0;
+                return result;
+            }
+        }
+
+        template Floating(T)
+        if (is(T == float) || is(T == double) || is(T == real))
+        {
+          pure nothrow @safe:
+            int compare(T d1, T d2)
+            {
+                if (d1 != d1 || d2 != d2) // if either are NaN
+                {
+                    if (d1 != d1)
+                    {
+                        if (d2 != d2)
+                            return 0;
+                        return -1;
+                    }
+                    return 1;
+                }
+                return (d1 == d2) ? 0 : ((d1 < d2) ? -1 : 1);
+            }
+        }
+
+        template Array(T)
+        if (is(T ==  float) || is(T ==  double) || is(T ==  real) ||
+            is(T == cfloat) || is(T == cdouble) || is(T == creal))
+        {
+          pure nothrow @safe:
+            int compare(T[] s1, T[] s2)
+            {
+                size_t len = s1.length;
+                if (s2.length < len)
+                    len = s2.length;
+                for (size_t u = 0; u < len; u++)
+                {
+                    if (int c = Floating!T.compare(s1[u], s2[u]))
+                        return c;
+                }
+                if (s1.length < s2.length)
+                    return -1;
+                else if (s1.length > s2.length)
+                    return 1;
+                return 0;
+            }
+        }
+
+        static if (is(U1 == ifloat)) alias F = float;
+        else static if (is(U1 == idouble)) alias F = double;
+        else static if (is(U1 == ireal)) alias F = real;
+        else alias F = U1;
+
+        return () @trusted { return Array!F.compare(cast(F[])s1, cast(F[])s2); }();
+    }
+
+    // integral, char types
+    else
+    {
+        size_t len = s1.length;
+
+        if (s2.length < len)
+            len = s2.length;
+
+        for (size_t u = 0; u < len; u++)
```

Comment:
> Think it over - that wouldn't work (unless you make the import public, which I understand you'd want to avoid).

@andralex I've already suggested `public import` twice.

What's even more concerning than the code side, is that it adds additional public symbols to the module that is by default imported. That can cause conflicts with existing symbols. What if I already have a function with the same name and a matching signature, will the compiler use that one instead? Or will it use a fully qualified name when invoking it?

What would be even better, and solve both of these problems, is to put the internal symbols in a separate module. When the compiler needs to access an internal symbol it would add a import in the lowering process for that module. It would use a renamed import to a compiler generated symbol. Example:

```d
import __someCompilerGeneratedSymbol123 = rt.internal;
__someCompilerGeneratedSymbol123._adCmpT(a, b);
```

It could even, if necessary, combine that with a selective import to only import what's exactly needed.